src/components/course/enrollment/enrollactions.jsx

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable import/prefer-default-export */
+import React, { useContext } from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+
+import { AppContext } from '@edx/frontend-platform/react';
+import { EnrollButtonCta } from './common';
+import { enrollLinkClass } from '../data/constants';
+import ToVoucherRedeemPage from './ToVoucherRedeemPage';
+import ToCoursewarePage from './ToCoursewarePage';
+import ToEcomBasketPage from './ToEcomBasketPage';
+
+/**
+ * Exports a number of UI components that represent the correct enroll button behavior for each
+ * scenario supported. Use the correct one based on the scenario.
+ */
+
+// Exported components
+
+// Disabled enroll
+const EnrollBtnDisabled = ({ enrollLabel }) => (
+  <EnrollButtonCta
+    enrollLabel={enrollLabel}
+    as="div"
+    className="btn btn-light btn-block disabled"
+  />
+);
+
+EnrollBtnDisabled.propTypes = {
+  enrollLabel: PropTypes.shape.isRequired,
+};
+
+// Data sharing consent
+const ToDataSharingConsentPage = ({ enrollLabel, enrollmentUrl }) => (
```

Comment:
Should these components either all be in new files or all included here in this file for consistency?